src/api/v1/event/index.js

```diff
@@ -130,4 +132,25 @@ router.route('/')
     }).catch(next);
   });
 
+/**
+ * Uploads a profile picture for a given event. Requires admin access
+ */
+router.post('/picture', Storage.bufferImageBlob(Media.mediaTypes.EVENT_COVER), async (req, res, next) => {
```

Comment:
I think something like `POST /picture/:uuid` would be a bit cleaner. We can also get rid of the UUID check in the controller then.

Reply:
I'm not entirely sure how authentication works, but for some reason when I change this event route to `POST /picture/:uuid`, `req.user` becomes undefined (preventing me from checking admin status) unless I specify the `authenticated` middleware in the route parameters. With `POST /picture/`, `req.user` gets populated just fine.

The weirder thing is with the user `POST /picture/:uuid` route, `req.user` populates just fine (probably since it's authenticated by default).

Do you know why authentication is different for routes with/without params? Or why the event route in particular is failing? I can't seem to find an explanation online so I think it's likely to do with how the other event routes are setup, not too sure

Reply:
Yeah, so because some `/event` routes are public (namely, `/past`, `/future`, and maybe the individual `GET /:uuid` route), `/event` is declared as part of the public API [here](https://github.com/acmucsd/membership-portal/blob/master/src/api/v1/index.js). The `authenticated` middleware must then be run on individual `/event` routes. If you take a look at the other routes, you'll see that they're all preceded by a block of code that runs the `authenticated` middleware (which populates `req.user`) and adds an admin check. To add authentication to an individual route, you can just add `authenticated` to the route, e.g.
```javascript
router.post('/route', authenticated, async (req, res, next) => { ... })
```